flake: TestTxClientTestSuite fails with "tx not found" after SubmitTx

## Summary

`TestTxClientTestSuite` failed on `main` with `rpc error: code = NotFound desc = tx not found` in multiple subtests. All failures share the same pattern: the test calls `suite.serviceClient.GetTx(...)` immediately after `SubmitTx`/`ConfirmTx` returns, and the Cosmos SDK tx indexer has not yet indexed the transaction.

## Failing run

- Workflow: Post-merge CI
- Run: https://github.com/celestiaorg/celestia-app/actions/runs/24529077171/job/71707175371
- Commit: `e0f7fa9de0f2647d268c4e4ab20874796d7f72db` (feat(fibre): replace ConstantValsetClient with TTL-based CachingClient)
- Job: `test / go-test (github.com/celestiaorg/celestia-app/v9/pkg/user)`

## Failures

```
--- FAIL: TestTxClientTestSuite (40.38s)
    --- FAIL: TestTxClientTestSuite/TestConfirmTx (5.03s)
        --- FAIL: TestTxClientTestSuite/TestConfirmTx/should_error_when_tx_is_found_with_a_non-zero_error_code (1.01s)
            Error Trace: pkg/user/tx_client_test.go:338
            Error:       rpc error: code = NotFound desc = tx not found: A4BAFD3AD32355823330055EE227F1B15F8B297899F496436AD81C24B66C5D81
    --- FAIL: TestTxClientTestSuite/TestGasConsumption (1.01s)
            Error Trace: pkg/user/tx_client_test.go:610
            Error:       Not equal
    --- FAIL: TestTxClientTestSuite/TestSubmitTx (6.03s)
        --- FAIL: TestTxClientTestSuite/TestSubmitTx/submit_tx_without_provided_fee_and_gas_limit (1.00s)
            Error Trace: pkg/user/tx_client_test.go:182
            Error:       rpc error: code = NotFound desc = tx not found: 1963892157B0D765AD7FDF4904B1968881524EC1BAD7E816FF2AC71A44BA3918
```

## Root cause (hypothesis)

`TxClient.ConfirmTx` polls CometBFT's in-memory `TxStatus` RPC, which returns `TxStatusCommitted` as soon as a transaction is included in a committed block. However, the Cosmos SDK tx service (`ServiceClient.GetTx`) reads from the tx indexer, which runs asynchronously and can briefly lag behind the commit. Tests that call `GetTx` immediately after `SubmitTx`/`ConfirmTx` are therefore racy.

## Proposed fix

Wrap the test-side `GetTx` calls (and `VerifyTxResponse`) with a short polling helper that retries on `NotFound` until the indexer catches up.

## Related

- #5667 (closed)
- #4909 (closed)

--- a/pkg/user/tx_client_test.go
+++ b/pkg/user/tx_client_test.go
@@ -65,7 +65,7 @@ func (suite *TxClientTestSuite) TestSubmitPayForBlob() {
 	t.Run("submit blob without provided fee and gas limit", func(t *testing.T) {
 		resp, err := suite.txClient.SubmitPayForBlob(subCtx, blobs)
 		require.NoError(t, err)
-		getTxResp, err := suite.serviceClient.GetTx(subCtx, &sdktx.GetTxRequest{Hash: resp.TxHash})
+		getTxResp, err := utils.GetTxWithRetry(subCtx, suite.serviceClient, resp.TxHash)
 		require.NoError(t, err)
 		require.EqualValues(t, 0, resp.Code)
 		require.Greater(t, getTxResp.TxResponse.GasWanted, int64(0))
@@ -76,7 +76,7 @@ func (suite *TxClientTestSuite) TestSubmitPayForBlob() {
 		gas := user.SetGasLimit(1e6)
 		resp, err := suite.txClient.SubmitPayForBlob(subCtx, blobs, fee, gas)
 		require.NoError(t, err)
-		getTxResp, err := suite.serviceClient.GetTx(subCtx, &sdktx.GetTxRequest{Hash: resp.TxHash})
+		getTxResp, err := utils.GetTxWithRetry(subCtx, suite.serviceClient, resp.TxHash)
 		require.NoError(t, err)
 		require.EqualValues(t, 0, resp.Code)
 		require.EqualValues(t, getTxResp.TxResponse.GasWanted, 1e6)
@@ -98,7 +98,7 @@ func (suite *TxClientTestSuite) TestSubmitPayForBlob() {
 		require.NotEmpty(t, accountName, "could not find a non-default account")
 		resp, err := suite.txClient.SubmitPayForBlobWithAccount(subCtx, accountName, blobs, user.SetFee(1e6), user.SetGasLimit(1e6))
 		require.NoError(t, err)
-		getTxResp, err := suite.serviceClient.GetTx(subCtx, &sdktx.GetTxRequest{Hash: resp.TxHash})
+		getTxResp, err := utils.GetTxWithRetry(subCtx, suite.serviceClient, resp.TxHash)
 		require.NoError(t, err)
 		require.EqualValues(t, 0, resp.Code)
 		require.EqualValues(t, getTxResp.TxResponse.GasWanted, 1e6)
@@ -146,7 +146,7 @@ func TestSubmitPayForBlobWithEstimatorService(t *testing.T) {
 	require.Equal(t, abci.CodeTypeOK, resp.Code)
 
 	serviceClient := sdktx.NewServiceClient(ctx.GRPCClient)
-	getTxResp, err := serviceClient.GetTx(ctx.GoContext(), &sdktx.GetTxRequest{Hash: resp.TxHash})
+	getTxResp, err := utils.GetTxWithRetry(ctx.GoContext(), serviceClient, resp.TxHash)
 	require.NoError(t, err)
 
 	require.Equal(t, int64(70000), getTxResp.TxResponse.GasWanted)
@@ -178,7 +178,7 @@ func (suite *TxClientTestSuite) TestSubmitTx() {
 		resp, err := suite.txClient.SubmitTx(suite.ctx.GoContext(), []sdk.Msg{msg})
 		require.NoError(t, err)
 		require.Equal(t, abci.CodeTypeOK, resp.Code)
-		getTxResp, err := suite.serviceClient.GetTx(suite.ctx.GoContext(), &sdktx.GetTxRequest{Hash: resp.TxHash})
+		getTxResp, err := utils.GetTxWithRetry(suite.ctx.GoContext(), suite.serviceClient, resp.TxHash)
 		require.NoError(t, err)
 		require.Greater(t, getTxResp.TxResponse.GasWanted, int64(0))
 	})
@@ -191,7 +191,7 @@ func (suite *TxClientTestSuite) TestSubmitTx() {
 		resp, err := suite.txClient.SubmitTx(suite.ctx.GoContext(), []sdk.Msg{msg}, gasLimitOption)
 		require.NoError(t, err)
 		require.Equal(t, abci.CodeTypeOK, resp.Code)
-		getTxResp, err := suite.serviceClient.GetTx(suite.ctx.GoContext(), &sdktx.GetTxRequest{Hash: resp.TxHash})
+		getTxResp, err := utils.GetTxWithRetry(suite.ctx.GoContext(), suite.serviceClient, resp.TxHash)
 		require.NoError(t, err)
 		require.EqualValues(t, int64(gasLimit), getTxResp.TxResponse.GasWanted)
 	})
@@ -206,7 +206,7 @@ func (suite *TxClientTestSuite) TestSubmitTx() {
 		resp, err := suite.txClient.SubmitTx(suite.ctx.GoContext(), []sdk.Msg{msg}, feeOption, gasLimitOption)
 		require.NoError(t, err)
 		require.Equal(t, abci.CodeTypeOK, resp.Code)
-		getTxResp, err := suite.serviceClient.GetTx(suite.ctx.GoContext(), &sdktx.GetTxRequest{Hash: resp.TxHash})
+		getTxResp, err := utils.GetTxWithRetry(suite.ctx.GoContext(), suite.serviceClient, resp.TxHash)
 		require.NoError(t, err)
 		require.EqualValues(t, int64(gasLimit), getTxResp.TxResponse.GasWanted)
 	})
@@ -334,7 +334,7 @@ func (suite *TxClientTestSuite) TestConfirmTx() {
 		require.Equal(t, resp.TxHash, err.(*user.ExecutionError).TxHash)
 
 		// Compare it to the getTx response
-		getTxResp, getTxErr := suite.serviceClient.GetTx(suite.ctx.GoContext(), &sdktx.GetTxRequest{Hash: resp.TxHash})
+		getTxResp, getTxErr := utils.GetTxWithRetry(suite.ctx.GoContext(), suite.serviceClient, resp.TxHash)
 		require.NoError(t, getTxErr)
 		// This is a workaround because they are different types
 		require.Contains(t, err.(*user.ExecutionError).ErrorLog, getTxResp.TxResponse.RawLog)
@@ -609,7 +609,7 @@ func (suite *TxClientTestSuite) TestGasConsumption() {
 	amountDeducted := balanceBefore - balanceAfter - utiaToSend
 	require.Equal(t, int64(fee), amountDeducted)
 
-	res, err := suite.serviceClient.GetTx(suite.ctx.GoContext(), &sdktx.GetTxRequest{Hash: resp.TxHash})
+	res, err := utils.GetTxWithRetry(suite.ctx.GoContext(), suite.serviceClient, resp.TxHash)
 	require.NoError(t, err)
 
 	// verify that the amount deducted does not depend on the actual gas used.

--- a/pkg/user/utils/txclient_utils.go
+++ b/pkg/user/utils/txclient_utils.go
@@ -2,6 +2,7 @@ package utils
 
 import (
 	"context"
+	"fmt"
 	"testing"
 	"time"
 
@@ -15,7 +16,13 @@ import (
 	sdktypes "github.com/cosmos/cosmos-sdk/types"
 	sdktx "github.com/cosmos/cosmos-sdk/types/tx"
 	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
+
+// indexerWaitTimeout bounds how long GetTxWithRetry waits for the tx indexer
+// to index a committed tx before giving up.
+const indexerWaitTimeout = 10 * time.Second
 
 func SetupTxClient(
 	t *testing.T,
@@ -53,6 +60,36 @@ func SetupTxClientWithDefaultParams(t *testing.T, opts ...user.Option) (encoding
 	return SetupTxClient(t, 0, 128, 8388608, opts...) // no ttl and 8MiB block size
 }
 
+// GetTxWithRetry polls ServiceClient.GetTx until the tx is indexed or
+// indexerWaitTimeout elapses. The CometBFT TxStatus RPC (used by
+// TxClient.ConfirmTx) and the Cosmos SDK tx indexer (used by
+// ServiceClient.GetTx) are updated asynchronously after a block is
+// committed, so GetTx can return NotFound for a tx that ConfirmTx has just
+// reported as committed. Tests should prefer this helper whenever they need
+// to look up a tx by hash immediately after submitting it.
+func GetTxWithRetry(ctx context.Context, serviceClient sdktx.ServiceClient, txHash string) (*sdktx.GetTxResponse, error) {
+	ctx, cancel := context.WithTimeout(ctx, indexerWaitTimeout)
+	defer cancel()
+
+	ticker := time.NewTicker(50 * time.Millisecond)
+	defer ticker.Stop()
+
+	for {
+		resp, err := serviceClient.GetTx(ctx, &sdktx.GetTxRequest{Hash: txHash})
+		if err == nil {
+			return resp, nil
+		}
+		if status.Code(err) != codes.NotFound {
+			return nil, err
+		}
+		select {
+		case <-ctx.Done():
+			return nil, fmt.Errorf("waiting for tx %s to be indexed: %w", txHash, ctx.Err())
+		case <-ticker.C:
+		}
+	}
+}
+
 func VerifyTxResponse(
 	t *testing.T,
 	ctx context.Context,
@@ -77,7 +114,7 @@ func VerifyTxResponse(
 		require.FailNowf(t, "unexpected type", "unsupported confirmTxResp type: %T", confirmTxResp)
 	}
 
-	getTxResp, err := serviceClient.GetTx(ctx, &sdktx.GetTxRequest{Hash: expTxHash})
+	getTxResp, err := GetTxWithRetry(ctx, serviceClient, expTxHash)
 	require.NoError(t, err)
 
 	txResp := getTxResp.TxResponse

